UaParser.cpp

```diff
@@ -11,7 +11,7 @@
 #include <boost/algorithm/string.hpp>
 #include <yaml-cpp/yaml.h>
 
-namespace {
+namespace uap_cpp {
```

Comment:
Why not leave these functions in an anonymous namespace? They're not part of the API and thus should have internal linkage.

Reply:
Because it will require "using namespace" then, which is not a big deal in this given case, but is forbidden in general (e.g. https://google.github.io/styleguide/cppguide.html#Namespaces).

Reply:
It wouldn't. You'd only have to qualify `Device` and `Agent` (as `uap_cpp::Device` and `uap_cpp::Agent`, respectively) in three free functions: `parse_device_impl`, `parse_browser_impl` and `parse_os_impl`. Perfectly fine and doesn't export functions which we don't want exported. Placing them in `uap_cpp` only saves a negligible amount of typing. I'm going to merge your patch anyway, thanks!